docs/pre-commit.md

```diff
@@ -0,0 +1,109 @@
+# Secret leak prevention with pre-commit
```

Comment:
A general overall comment, if an error is detected say with a json file for example, can we list out the file? 

For example

```bash
check json...............................................................Failed
- hook id: check-json
- exit code: 1
- file: file a.json
- file file b.json
```